### PR TITLE
Fix #3321: commit tag shortener

### LINKED ISSUE #3321: Short commit references not turning into links.
See https://git.osgeo.org/gogs/geos/php-geos/issues/6#issuecomment-182 and subsequent comment.
The first comment used ca42d272d8 as a commit reference, the second used the full hash and the full url.
Only the second comment turned both references into links, while the first did not.

### PR DESCRIPTION
**Pull request by 0xbaadf00d**
Short commit references now working with a 7 to 40 SHA length'  (Issue #3321) and base.ShortSha() function has been modified to uniformize short SHA's representation (length: 7).

### CODE REVIEW

File: modules/base/tool.go [hunk 1]
```diff
@@ -45,8 +45,8 @@ func EncodeSha1(str string) string {
 }
 
 func ShortSha(sha1 string) string {
-	if len(sha1) == 40 {
```
Comment (Unknwon) [inline modules/base/tool.go]: We can allow 7-char SHA, does not mean we want to show with 7-char in default, please remove this change.

File: modules/markdown/markdown_test.go [hunk 2]
```diff
@@ -290,8 +290,8 @@ func TestMarkdown(t *testing.T) {
 		Convey("To the internal issue tracker", func() {
 			Convey("It should correctly convert URLs", func() {
 				testCases := []string{
-					"http://localhost:3000/user/project/commit/d8a994ef243349f321568f9e36d5c3f444b99cae", " <code><a href=\"http://localhost:3000/user/project/commit/d8a994ef243349f321568f9e36d5c3f444b99cae\">d8a994ef24</a></code>",
```
Comment (Unknwon) [inline modules/markdown/markdown_test.go]: We can allow 7-char SHA, does not mean we want to show with 7-char in default, please adjust this change.

File: modules/base/tool.go (hunk 1 above)
Comment (0xbaadf00d) [inline modules/base/tool.go]: I have done this to always have uniform short tag on all pages regardless of what users use (7-char, 8-char, ..., 40-char commit  SHA).

Ok if you want remove this; I will do needed modifications tonight.

Comment (Unknwon) [inline modules/base/tool.go]: Yes, please do the modification, we still want to show 10-char on the UI. Thought people could just use 7-char in other cases.

**Comment by Unknwon** [comment]
Thanks!

**Pull request merged**